src/socialMediaLinkSharing.js

```diff
@@ -11,40 +11,78 @@ import * as socialMediaActivity from "./socialMediaActivity.js";
 import * as linkResolution from "./linkResolution.js";
 
 /**
- * @constant {debugging.debuggingLogger}
- * @private
+ * @type {Events.Event<socialMediaShareCallback, socialMediaShareOptions>}
  */
-const debugLog = debugging.getDebuggingLog("socialMediaLinkSharing");
+export const onShare = events.createEvent({
```

Comment:
I think the boundary between `socialMediaActivity` and `socialMediaLinkSharing` should be a bit different from what it is. Currently, `socialMediaActivity` exposes functions for looking up tweets, posts, and subreddits, and `socialMediaLinkSharing` has to call those functions. It would be preferable if we could keep those functions internal to `socialMediaActivity` and only expose them if study developers eventually need them. The information that `socialMediaLinkSharing` has to obtain itself is also information that many callers to `socialMediaActivity` would want anyway. A possible approach would be a listener option for `socialMediaActivity` that indicates shared tweets and posts should be resolved, plus adding subreddit status to reddit events.